### PR TITLE
Adds basic retries to rest API requests for Python asyncio

### PR DESCRIPTION
**Pull request by rad-pat**
* This mimics the basic retries performed by urllib3 in the synchronous Python version

Tests have not been added as they were not present for the sync version. 
I raised the issue https://github.com/OpenAPITools/openapi-generator/issues/17011 and was told to open a PR as a fix looked fairly simple. Unfortunately if more is required, I don't presently have the bandwidth to help out.

### CODE REVIEW

**Comment by wing328** [comment]
thanks for the PR.

can you please review the test failure when you've time?
```
                    self._logger.debug(f"Retrying after response code: {response.status}")
                retry_wait = self._retry_options.get_timeout(attempt=current_attempt, response=response)
            except Exception as e:
>               if current_attempt >= self._retry_options.attempts:
E               TypeError: '>=' not supported between instances of 'int' and 'NoneType'
```

ref: https://github.com/OpenAPITools/openapi-generator/actions/runs/6807387600/job/18510156458?pr=17014

**Comment by wing328** [comment]
cc @cbornet (2017/09) @tomplus (2018/10) @krjakbrjak (2023/02) @fa0311 (2023/10) @multani (2023/10)

File: modules/openapi-generator/src/main/resources/python/asyncio/rest.mustache [hunk 1]
```diff
@@ -68,6 +71,20 @@ class RESTClientObject:
             trust_env=True
         )
 
+        self.retries = configuration.retries
```
Comment (multani) [inline modules/openapi-generator/src/main/resources/python/asyncio/rest.mustache:74]: I think you can just skip the `self.` part here, as you are now only using this `retries` attribute in the same function. No need to keep this value as an attribute of the class.

**Review by multani — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/python/asyncio/rest.mustache [hunk 2]
```diff
@@ -68,6 +71,20 @@ class RESTClientObject:
...
+                    factor=0.0,
+                    start_timeout=0.0,
+                    max_timeout=120.0
+                )
+            )
+        else:
+            self.retry_client = None
+
     async def close(self):
         await self.pool_manager.close()
```
Comment (fa0311) [inline modules/openapi-generator/src/main/resources/python/asyncio/rest.mustache:90]: From the aiohttp_retry documentation, it looks like RetryClient needs to be closed
https://github.com/inyutin/aiohttp_retry#examples-of-usage

Comment (rad-pat) [reply modules/openapi-generator/src/main/resources/python/asyncio/rest.mustache:90]: Closing the retry client just closes the session which is taken care of by the RESTClientObject.close() here.

Comment (multani) [reply modules/openapi-generator/src/main/resources/python/asyncio/rest.mustache:90]: It probably doesn't hurt to add the close() call and not to assume too much what RetryClient is doing though.
```suggestion
        if self.retry_client is not None:
            await self.retry_client.close()

```

File: modules/openapi-generator/src/main/resources/python/asyncio/rest.mustache [hunk 3]
```diff
@@ -168,7 +185,12 @@ class RESTClientObject:
                          declared content type."""
                 raise ApiException(status=0, reason=msg)
 
-        r = await self.pool_manager.request(**args)
+        if self.retry_client and method in ALLOW_RETRY_METHODS:
```
Comment (multani) [inline modules/openapi-generator/src/main/resources/python/asyncio/rest.mustache:188]:
```suggestion
        if self.retry_client is not None and method in ALLOW_RETRY_METHODS:
```

**Review by multani — CHANGES_REQUESTED** [review]
Can you check the last comments?

I think, it would be good to merge on my side after :+1:

**Review by multani — APPROVED** [review]
Looks good to me :bow:

**Pull request merged**